identity-account-storage/src/storage/stronghold.rs

```diff
@@ -389,11 +465,146 @@ fn sign_ed25519(client: &Client, payload: Vec<u8>, location: &KeyLocation) -> Re
 
   let signature: [u8; 64] = client
     .execute_procedure(procedure)
-    .map_err(|err| procedure_error::<procedures::Ed25519Sign>(location, err))?;
+    .map_err(|err| procedure_error::<procedures::Ed25519Sign>(vec![location.clone()], err))?;
 
   Ok(Signature::new(signature.into()))
 }
 
+pub(crate) async fn diffie_hellman(
+  client: &Client,
+  private_key: &KeyLocation,
+  public_key: [u8; X25519::PUBLIC_KEY_LENGTH],
+) -> Result<Location> {
+  let location: [u8; 32] = OsRng.sample(rand::distributions::Standard);
```

Comment:
As a performance improvement, we can switch all instances of `OsRng` back to `ThreadRng` now that we have a dedicated musl build container. Thanks @cycraig, for confirming.